WooCommerce/src/test/kotlin/com/woocommerce/android/ui/login/jetpack/wpcom/JetpackActivationMagicLinkRequestViewModelTest.kt

```diff
@@ -0,0 +1,136 @@
+import com.woocommerce.android.ui.login.MagicLinkFlow
+import com.woocommerce.android.ui.login.MagicLinkSource
+import com.woocommerce.android.ui.login.WPComLoginRepository
+import com.woocommerce.android.util.runAndCaptureValues
+import com.woocommerce.android.viewmodel.BaseUnitTest
+import com.woocommerce.android.viewmodel.ResourceProvider
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+import org.wordpress.android.login.MagicLinkFallbackButton
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class JetpackActivationMagicLinkRequestViewModelTest : BaseUnitTest() {
+    companion object {
+        private const val EMAIL = "email@example.com"
+        private val JetpackStatus = JetpackStatus(
+            isJetpackInstalled = true,
+            isJetpackConnected = false,
+            wpComEmail = null
+        )
+    }
+
+    private val analyticsTrackerWrapper: AnalyticsTrackerWrapper = mock()
+    private val wpComLoginRepository: WPComLoginRepository = mock()
+    private val resourceProvider: ResourceProvider = mock()
+
+    private lateinit var viewModel: JetpackActivationMagicLinkRequestViewModel
+
+    fun setup(
+        fallbackButton: MagicLinkFallbackButton,
+        requestEmailAtStart: Boolean = true
+    ) {
+        viewModel = JetpackActivationMagicLinkRequestViewModel(
+            JetpackActivationMagicLinkRequestFragmentArgs(
+                jetpackStatus = JetpackStatus,
+                emailOrUsername = EMAIL,
+                fallbackButton = fallbackButton,
+                requestAtStart = requestEmailAtStart,
+                isNewWpComAccount = false
+            ).toSavedStateHandle(),
+            resourceProvider,
+            wpComLoginRepository,
+            analyticsTrackerWrapper
+        )
+    }
+
+    @Test
+    fun `given request email at start is true, when view model is initialized, then request magic link`() =
+        testBlocking {
+            setup(MagicLinkFallbackButton.None, requestEmailAtStart = true)
+
+            verify(wpComLoginRepository).requestMagicLink(
+                emailOrUsername = EMAIL,
+                flow = MagicLinkFlow.SiteCredentialsToWPCom,
+                source = MagicLinkSource.JetpackConnection,
+                isSignup = false
+            )
+        }
+
+    @Test
+    fun `given request email at start is false, when view model is initialized, then do not request magic link`() =
+        testBlocking {
+            setup(MagicLinkFallbackButton.None, requestEmailAtStart = false)
+
+            verify(wpComLoginRepository, never()).requestMagicLink(
+                emailOrUsername = EMAIL,
+                flow = MagicLinkFlow.SiteCredentialsToWPCom,
+                source = MagicLinkSource.JetpackConnection,
+                isSignup = false
+            )
```

Comment:
Since we are checking `never` condition, I suggest having general parameters here. Because, in the future, we might change `SiteCredentialsToWPCom` to something else in the code. Then this test case may give false positive.
```suggestion
            verify(wpComLoginRepository, never()).requestMagicLink(any(), any(), any(), any())
```